hela/posteriors.py

```diff
@@ -179,6 +180,46 @@ def __init__(self, samples, weights):
         self.samples = samples
         self.weights = weights
 
+    def print_percentiles(self, names):
+        """
+        Print the median and the +/- 1 sigma posterior values.
+
+        Parameters
+        ----------
+        names
+
+        Returns
+        -------
+
+        """
+        posterior_ranges = self.data_ranges()
+        for name_i, pred_range_i in zip(names, posterior_ranges):
+            print("Prediction for {}: {:.3g} "
+                  "[+{:.3g} -{:.3g}]".format(name_i, *pred_range_i))
+
+    def data_ranges(self, percentiles=(50, 16, 84)):
+        """
+        Return posterior ranges.
+
+        Parameters
+        ----------
+        posterior : `~numpy.ndarray`
+        percentiles : tuple
+
+        Returns
+        -------
+        ranges : `~numpy.ndarray`
+        """
+        values = wpercentile(self.samples, self.weights,
+                             percentiles, axis=0)
+        ranges = np.array(
+            [values[0], values[2] - values[0], values[0] - values[1]])
+        return ranges.T
+
+    def plot_posterior_matrix(self, dataset):
```

Comment:
As in `print_percentiles`, this function performs I/O in the class `Posterior`. Same things apply here.

Also, this function leads to a circular dependency in the code, since `posteriors.py` now depends on `plot.py` and vice versa.